### PR TITLE
Use std::chrono clock in profile timers

### PR DESCRIPTION
**Pull request by markdewing**
Replace timer based on gettimeofday with std::chrono timer (system_clock) in the profile timers.

Reasons:
1. I would like to add the event profile code, and it works better with std::chrono timers
2. Storing the start time as a double (since the epoch) loses precision - the elapsed time ends up being a small difference between large numbers.  This isn't as major an issue with aggregated profile data, but is noticeable in the event data.
3. Moves to a C++ standard API

The runtime manager was not switched in this PR, but could be changed in the future (and the the FakeCPUClock removed)

The test uses "chrono_literals" ( https://en.cppreference.com/w/cpp/header/chrono#Literals ), which uses C++ 11 user-defined literals ( https://en.cppreference.com/w/cpp/language/user_literal ).   The "1.1s" literal syntax can be a little disorienting if one has not encountered it before.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New implementation of existing functionality
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Ye. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- N/A. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Utilities/Clock.h [hunk 1]
```diff
@@ -19,9 +19,33 @@
...
+#include <chrono>
 
 namespace qmcplusplus
 {
+
+
+// Implements a std::chrono clock
+// See https://github.com/korfuri/fake_clock
+
+class FakeChronoClock
```
Comment (ye-luo) [inline src/Utilities/Clock.h:31]: Can we use FakeChronoClock to replace all FakeCPUClock?

Comment (markdewing) [reply src/Utilities/Clock.h:31]: The runtime manager was not switched in this PR, but could be changed in the future (and the the FakeCPUClock removed)
I would prefer not to put that change in this PR, in order to keep PR's smaller.

Comment (ye-luo) [reply src/Utilities/Clock.h:31]: Hopefully the switch will happen relatively soon.

File: src/Utilities/NewTimer.cpp [hunk 2]
```diff
@@ -101,19 +101,19 @@ void TimerType<CLOCK>::stop()
...
 
-      per_stack_total_time[current_stack_key] += elapsed;
+      per_stack_total_time[current_stack_key] += elapsed.count();
       per_stack_num_calls[current_stack_key] += 1;
 
       if (manager)
         manager->pop_timer(this);
     }
 #else
-    double elapsed = CLOCK()() - start_time;
```
Comment (ye-luo) [inline src/Utilities/NewTimer.cpp:115]: I disliked the wired `()()`, the new function is more readable.

**Review by ye-luo — APPROVED** [review]
Still need non-ANL approval

File: src/Utilities/Clock.h (hunk 1 above)
Comment (prckent) [reply src/Utilities/Clock.h:31]: Indeed.

**Comment by prckent** [comment]
Test this please

**Review by prckent — APPROVED** [review]

**Pull request merged**